code/game/machinery/computer/computer.dm

```diff
@@ -36,9 +40,41 @@
 	set_light(0)
 	visible_message("<span class='danger'>[src] grows dim, its screen barely readable.</span>")
 
+/*
+ * Reimp, flash the screen on and off repeatedly.
+ */
+/obj/machinery/computer/flicker()
+	if(flickering)
+		return FALSE
+
+	if(stat & (BROKEN|NOPOWER))
+		return FALSE
+
+	flickering = TRUE
+	INVOKE_ASYNC(src, /obj/machinery/computer/.proc/flicker_event)
+
+	return TRUE
+
+/*
+ * Proc to be called by invoke_async in the above flicker() proc.
+ */
+/obj/machinery/computer/proc/flicker_event()
+	var/amount = rand(5, 15)
+
+	for(var/i = 0; i < amount; i++)
+		force_no_power_icon_state = TRUE
+		update_icon()
+		sleep(rand(1, 3))
+
+		force_no_power_icon_state = FALSE
+		update_icon()
+		sleep(rand(1, 10))
+	update_icon()
+	flickering = FALSE
+
 /obj/machinery/computer/update_icon()
 	overlays.Cut()
-	if(stat & NOPOWER)
+	if(stat & NOPOWER || force_no_power_icon_state)
```

Comment:
I believe this needs parentheses around it, or you'll get some unintended behavior.
```suggestion
	if((stat & NOPOWER) || force_no_power_icon_state)
```